crates/router/src/core/payment_methods/cards.rs

```diff
@@ -654,7 +663,16 @@ pub async fn mock_get_card<'a>(
         card_fingerprint: locker_mock_up.card_fingerprint.into(),
         card_global_fingerprint: locker_mock_up.card_global_fingerprint.into(),
         merchant_id: Some(locker_mock_up.merchant_id),
-        card_number: Some(locker_mock_up.card_number.try_into().unwrap()),
+        card_number: match locker_mock_up
+            .card_number
+            .try_into()
+            .into_report()
+            .change_context(errors::VaultError::ResponseDeserializationFailed)
+            .attach_printable("Invalid card number format from the mock locker")
+        {
+            Ok(ccnum) => Some(ccnum),
+            Err(e) => return Err(e),
+        },
```

Comment:
You can perform a similar conversion here.

```suggestion
        card_number: locker_mock_up
            .card_number
            .try_into()
            .into_report()
            .change_context(errors::VaultError::ResponseDeserializationFailed)
            .attach_printable("Invalid card number format from the mock locker")
            .map(Some)?,
```

Reply:
good call! :+1: